acisfp_check/acis_obs.py

```diff
@@ -40,11 +40,83 @@ def who_in_fp(simpos=80655):
     return is_in_the_fp
 
 
-# ---------------------------------------------------------------------
-#
-# find_obsid_intervals
-#
-# ---------------------------------------------------------------------
+def fetch_ocat_data(obsid_list):
+    """
+    Take a list of obsids and return the following data from
+    the obscat for each: grating status, CCD count, if S3 is on,
+    and the number of expected counts
+
+    Parameters
+    ----------
+    obsid_list : list of ints
+        The obsids to get the obscat data from.
+
+    Returns
+    -------
+    A dict of NumPy arrays of the above properties.
+    """
+    import requests
+    from astropy.io import ascii
+    # The following uses a request call to the obscat which explicitly
+    # asks for text formatting so that the output can be ingested into
+    # an AstroPy table.
+    urlbase = "https://cda.harvard.edu/srservices/ocatDetails.do?format=text"
+    obsid_list = ",".join([str(obsid) for obsid in obsid_list])
+    params = {"obsid": obsid_list}
+    # First fetch the information from the obsid itself
+    got_table = True
+    try:
```

Comment:
Just a heads-up that we have a handy function to allow robustly retrying a function (e.g. a web query, file access) that might momentarily fail but succeed a few seconds later.
https://github.com/sot/ska_helpers/blob/0867570c7dd1b44d22c1b61b8b3b44875a47c3b1/ska_helpers/retry/api.py#L108
Maybe overkill or not useful in this context, but we've found this useful especially in accessing HDF5 files on the NetApp.

Reply:
Hi @taldcroft, do you have a good code example where this is used?

Reply:
I know of a couple examples at https://github.com/sot/kadi/blob/593d705c023a2d982d30fe1cb74ba44b4d6006e2/kadi/commands/commands.py#L46 and https://github.com/sot/kadi/blob/593d705c023a2d982d30fe1cb74ba44b4d6006e2/kadi/update_cmds.py#L328

Reply:
This was a good suggestion--I've implemented it now.